### DISCUSSION TITLE
Devcontainer for Fedify

### DISCUSSION THREAD

**Discussion by moreal** [fedify-dev/fedify#142] [Show and tell]
## 🙌🏻 Introduction

Hi, I tried to make devcontainer[^1] configuration and you can see [here][devcontainer-commit]. It does:

 - When building(creating) a new container, `deno`, `bun`, `node`, `pnpm`.
 - When attaching to the container, run `deno task codegen`[^2]
 - Same vscode extensions will be installed with `.vscode/extensions.json`[^3]
 
[devcontainer-commit]: https://github.com/moreal/fedify/commit/94554052a446b14b1df8232286310b9e6864b9f0

[^1]: https://containers.dev/
[^2]: In details, `deno task -c src/deno.json codegen; deno task -c cli/deno.json`. I added these instructions because CONTRIBUTING.md says it; https://github.com/dahlia/fedify/blob/451def15df40bf08d3d38830d3f132b22994090d/CONTRIBUTING.md?plain=1#L66
[^3]: https://code.visualstudio.com/docs/devcontainers/containers#_adding-an-extension-to-devcontainerjson

## 🤔 Why not pull request?

I think the devcontainer will be helpful to setup Fedify development environment for new contributors, and use as isolated development environment even for hard contributors. But I don't want to open this as PR, yet.

It is because I'm not sure about the below items:

 - ❓ How many people uses devcontainers? If it is not useful, it will be unnecessary item which spends management cost (e.g., package manager version bump).
 - ❓ Will it be really helpful for contributors? 🤔 

I wonder the opinion of maintainers and other contributors.

**Comment by dahlia** [comment]
Providing a dev container out of box would be a good idea! It would be really useful, especially for first-time contributors.

Also, about management cost: I don't believe it's that expensive—go for it!

**Reply by moreal** [reply]
Thanks for your feedback! I'll open PR!

**Reply by moreal** [reply]
The PR is here! https://github.com/dahlia/fedify/pull/143 🚀
